cryptoki/src/slot/mod.rs

```diff
@@ -163,121 +162,356 @@ impl From<CK_SLOT_INFO> for SlotInfo {
+    /// If `Some(Some(n))` the maximum number of sessions is `n`
+    pub fn max_session_count(&self) -> Option<Option<u64>> {
+        self.max_session_count
+    }
+
+    /// The number of sessions this application currently has open with the
+    /// token
+    pub fn session_count(&self) -> Option<u64> {
+        self.session_count
+    }
+
+    /// The maximum number of read/write sessions that can be opened with the
+    /// token at one time by a single application
+    /// If `None`, this information was unavailable.
+    /// If `Some(None)` there is no maximum, meaning the value is effectively infinite
+    /// If `Some(Some(n))` the maximum number of read/write sessions is `n`
+    pub fn max_rw_session_count(&self) -> Option<Option<u64>> {
+        self.max_rw_session_count
+    }
+
+    /// The number of read/write sessions this application currently has open
+    /// with the token
+    pub fn rw_session_count(&self) -> Option<u64> {
+        self.rw_session_count
+    }
+
+    /// The maximum length in bytes of the PIN
+    pub fn max_pin_length(&self) -> usize {
+        self.max_pin_len
     }
-}
 
-impl From<TokenInfo> for CK_TOKEN_INFO {
-    fn from(token_info: TokenInfo) -> Self {
-        *token_info
+    /// The minimum length in bytes of the PIN
+    pub fn min_pin_length(&self) -> usize {
+        self.min_pin_len
+    }
+
+    /// The total amount of memory on the token (in bytes) in which public
+    /// objects may be stored
+    /// Returns `None` if this information is unavailable
+    pub fn total_public_memory(&self) -> Option<usize> {
+        self.total_public_memory
+    }
+
+    /// The amount of free (unused) emmeory on the token (in bytes) for public
+    /// objects
+    /// Returns `None` if this information is unavailable
+    pub fn free_public_memory(&self) -> Option<usize> {
+        self.free_public_memory
+    }
+
+    /// The total amount of memory on the token (in bytes) in which private
+    /// objects may be stored
+    /// Returns `None` if this information is unavailable
+    pub fn total_private_memory(&self) -> Option<usize> {
+        self.total_private_memory
+    }
+
+    /// The amount of free (unused) emmeory on the token (in bytes) for private
+    /// objects
+    /// Returns `None` if this information is unavailable
+    pub fn free_private_memory(&self) -> Option<usize> {
+        self.free_private_memory
+    }
+
+    /// The version number of the hardware
+    pub fn hardware_version(&self) -> Version {
+        self.hardware_version
+    }
+
+    /// The version number of the firmware
+    pub fn firmware_version(&self) -> Version {
+        self.firmware_version
+    }
+
+    /// The current datetime with resolution in seconds
+    ///
+    /// Returns None if the token is not equipped with a clock (i.e.,
+    /// `self.clock_on_token() == false`)
+    pub fn utc_time(&self) -> Option<&str> {
+        // TODO
```

Comment:
what is there todo?

Reply:
Ah, that's to myself, and is just part of my personal workflow. It would have (will be) removed before marking the PR ready for merge. I leave these behind for myself as I work (not always with explanatory text if the context is enough) when I have something I want to return to but not get sidetracked by right then or even have to switch contexts to a separate document. Any time I think I've reached my intended goal, I'll do a sweep for them, and they only occasionally make it into WIP commits.

In this case, I'd like to replace these raw bytes with an existing, functional date-time type TBD. But that's a completely different PR. So this tag is just going to get pulled out into a better note elsewhere.